gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -98,6 +102,19 @@ public void setProp(String key, Object value) {
   }
 
   /**
+   * Appends the input value to a list property that can be retrieved with {@link #getPropAsList}.
+   * @param key property key
+   * @param value property value (should not include commas, which are used as separators)
```

Comment:
One thing we can do is to add a new method `getPropAsList(String separator, String key)`, which allows users to specify a separator, and simply call `getPropAsList(",", key)` in `getPropAsList(key)` as a default and make clear in the Javadoc for this default that it is expecting comma as separator.